### PR TITLE
LWDEV-10450 fix: do not use other beans while registering bean definition dynamically

### PR DESCRIPTION
**Pull request by barsukovskii**

### CODE REVIEW

File: src/main/kotlin/com/lykke/trade/volume/monitoring/service/spring/IncomingEventProcessConfig.kt [hunk 1]
```diff
@@ -271,7 +280,7 @@ class IncomingEventProcessConfig : BeanFactoryPostProcessor {
 
         val constructorArgumentValues = ConstructorArgumentValues()
         constructorArgumentValues.addIndexedArgumentValue(0, index)
-        constructorArgumentValues.addIndexedArgumentValue(1, factory.getBean("incomingEventProcessThreadPool"))
+        constructorArgumentValues.addIndexedArgumentValue(1, RuntimeBeanReference("incomingEventProcessThreadPool"))
```
Comment (olpapchenko) [inline src/main/kotlin/com/lykke/trade/volume/monitoring/service/spring/IncomingEventProcessConfig.kt]: extract bean name to constant

File: src/main/kotlin/com/lykke/trade/volume/monitoring/service/spring/ConfigFactory.kt [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+        fun getConfig(environment: Environment): Config {
+            if (config == null) {
+                config = ConfigInitializer.initConfig(getConfigUrl(environment), classOfT = Config::class.java)
+            }
+            return config as Config
+        }
+
+        private fun getConfigUrl(environment: Environment): String {
+            return if (environment.acceptsProfiles(Profiles.of(LOCAL_CONFIG_PROFILE)))
+                "local"
```
Comment (olpapchenko) [inline src/main/kotlin/com/lykke/trade/volume/monitoring/service/spring/ConfigFactory.kt]: suggest adding here constant

**Review by olpapchenko — APPROVED** [review]

File: src/main/kotlin/com/lykke/trade/volume/monitoring/service/spring/IncomingEventProcessConfig.kt (hunk 1 above)
Comment (barsukovskii) [reply src/main/kotlin/com/lykke/trade/volume/monitoring/service/spring/IncomingEventProcessConfig.kt]: done

File: src/main/kotlin/com/lykke/trade/volume/monitoring/service/spring/ConfigFactory.kt (hunk 2 above)
Comment (barsukovskii) [reply src/main/kotlin/com/lykke/trade/volume/monitoring/service/spring/ConfigFactory.kt]: done

**Pull request merged**